utils/python/CIME/XML/entry_id.py

```diff
@@ -151,36 +182,71 @@ def get_elements_from_child_content(self, childname, childcontent):
         nodes = self.get_nodes("entry")
         elements = []
         for node in nodes:
-            childnode = self.get_node(childname,root=node)
+            childnode = self.get_optional_node(childname,root=node)
+            expect(childnode is not None,"No childname %s for id %s"%(childname,node.get("id")))
             content = childnode.text
             if content == childcontent:
-                elements.append(node)
+                elements.append(deepcopy(node))
```

Comment:
This concerns me. I don't understand how deepcopy works with XML elements, but for standing python containers, I would not approve this change. My specific concern is that any modifications to the deepcopied nodes would not be reflected in the XML (either in memory or on file) because you are only modifying a copy. If you need children of nodes, I'd prefer some kind of recursion.